_posts/2016-10-11-approaching-tdd-outside-android-iii.md

```diff
@@ -0,0 +1,103 @@
+<li><span style="color: #388e3c; padding-right: 5px;">●</span><a href="https://github.com/CarlosMChica/AndroidBankKata/commit/8a262bf34a4f50d900996b22ac1465c24a219ffc">Green</a> - As described before, the production code for the StatementFormatter takes the lines from the statement, sort them in inverse chronological order and map them them to ViewStatementLines.</li>
+<li><span style="color: #1976d2; padding-right: 5px;">●</span><a href="https://github.com/CarlosMChica/AndroidBankKata/commit/2b0755e733a7ee570f8fcd
+a46d5ca6975f50f5a5">Refactor</a> - We extracted a method that maps a line in the production code and do some clean up in the test code.</li>
+</ul>
+
+Once again, we run the acceptance test to check the progress and now it guide us to the lines method in the Statement class (The one that we just mocked in the previous inner loop to implement the StatementFormatter). Let's get rid of the exception and jump into the next inner loop.
+
+<ul style="display: inline-block; list-style: none; text-align: left;">
+<li><span style="color: #d32f2f; padding-right: 5px;">●</span><a href="https://github.com/CarlosMChica/AndroidBankKata/commit/e53c6ba6bcec339af7cdfaab148c24fa8aa8ac55">Red</a> - Statement has to map every transaction to a StatementLine that contains amount, date and running balance.
+As in the case of the TransactionRepository, Statement is a leaf node. Therefore, it is tested using the classicist approach by asserting the state of the StatementLines returned in the lines method.</li>
+<li><span style="color: #388e3c; padding-right: 5px;">●</span><a href="https://github.com/CarlosMChica/AndroidBankKata/commit/67631395b147b31bd1887fdad8485a298e6eff12">Green</a> - The implementation just map all Statement's transactions to StatementLines, calculating their current balance.</li>
+<li><span style="color: #1976d2; padding-right: 5px;">●</span><a href="https://github.com/CarlosMChica/AndroidBankKata/commit/8f4e26cd2f9387c9f8d0102a026f4f6ef58bcc7d" >Refactor</a> - We extracted some methods in the production code and improved test readability.</li>
+</ul>
+
+We are now done with the Statement class. Let's run the acceptance test again to find out that the next class that needs to be implemented is the show method of the <b>ShowStatementActivity</b>.
+
+<ul style="display: inline-block; list-style: none; text-align: left;">
+<li><span style="color: #d32f2f; padding-right: 5px;">●</span><a href="https://github.com/CarlosMChica/AndroidBankKata/commit/b325378396244bcfb04bf2fb14eaf76ea40a5a0d">Red</a> - During most part of the implementation we were not dealing with any Android code, but at this level we moved back to the to UI layer and, as the code is running inside an Activity, we have to write the unit test using Espresso. Here we need to test that once the show method is called, the RecyclerView contains a list of rows representing the information contained in the given ViewStatementLines.</li>
```

Comment:
During most of the implementation we were not dealing with any Android code. At this level we moved back to the to UI layer and, as the code is running inside an Activity, we have to write the unit test using Espresso.